topics/admin/tutorials/ansible-galaxy/tutorial.md

```diff
@@ -410,7 +420,7 @@ The configuration is quite simple thanks to the many sensible defaults that are
 >    1. `admin_users` to the email address you will use with this Galaxy.
 >    2. `brand` to something fun!
 >    3. `database_connection` to point to the database you setup earlier (`postgresql:///galaxy?host=/var/run/postgresql`).
->    4. `file_path` to a place to store data, `/data` for this lesson.
+>    4. `file_path` to a place to store data, `/data` is fine for this lesson which sets up a single-node Galaxy. If you have separate compute machines, this will normally need to be storage shared between the Galaxy node and compute nodes.
 >    5. `check_migrate_tools` must be set to `false` due to a new installation of Galaxy.
 >    6. `shed_tool_data_dir` to `{{ galaxy_mutable_data_dir }}/tool-data`, so that when tools are installed, due to privilege separation, this will happen in a directory Galaxy can actually write into.
```

Comment:
```suggestion
>    6. `shed_tool_data_path` to `{{ galaxy_mutable_data_dir }}/tool-data`, so that when tools are installed, due to privilege separation, this will happen in a directory Galaxy can actually write into.
```

Reply:
`shed_tool_data_path` appears with the wrong name (`shed_tool_data_dir`) in other 2 places below.
@natefoo It feels wrong that this needs to be set at all, right?

Reply:
And this is still not enough to fix the installation of bwa tool:
```
Feb 21 04:31:56 gcc-6 uwsgi[25895]: galaxy.web.framework.decorators ERROR 2020-02-21 04:31:56,092 [p:26003,w:1,m:0] [uWSGIWorker1Core3] Uncaught exception in exposed API method:
Feb 21 04:31:56 gcc-6 uwsgi[25895]: Traceback (most recent call last):
Feb 21 04:31:56 gcc-6 uwsgi[25895]:   File "/srv/galaxy/server/lib/galaxy/web/framework/decorators.py", line 282, in decorator
Feb 21 04:31:56 gcc-6 uwsgi[25895]:     rval = func(self, trans, *args, **kwargs)
Feb 21 04:31:56 gcc-6 uwsgi[25895]:   File "/srv/galaxy/server/lib/galaxy/webapps/galaxy/api/tool_shed_repositories.py", line 139, in install_repository_revision
Feb 21 04:31:56 gcc-6 uwsgi[25895]:     payload)
Feb 21 04:31:56 gcc-6 uwsgi[25895]:   File "/srv/galaxy/server/lib/galaxy/tool_shed/galaxy_install/install_manager.py", line 718, in install
Feb 21 04:31:56 gcc-6 uwsgi[25895]:     install_options
Feb 21 04:31:56 gcc-6 uwsgi[25895]:   File "/srv/galaxy/server/lib/galaxy/tool_shed/galaxy_install/install_manager.py", line 812, in __initiate_and_install_repositories
Feb 21 04:31:56 gcc-6 uwsgi[25895]:     return self.install_repositories(tsr_ids, decoded_kwd, reinstalling=False, install_options=install_options)
Feb 21 04:31:56 gcc-6 uwsgi[25895]:   File "/srv/galaxy/server/lib/galaxy/tool_shed/galaxy_install/install_manager.py", line 859, in install_repositories
Feb 21 04:31:56 gcc-6 uwsgi[25895]:     tool_panel_section_mapping=tool_panel_section_mapping)
Feb 21 04:31:56 gcc-6 uwsgi[25895]:   File "/srv/galaxy/server/lib/galaxy/tool_shed/galaxy_install/install_manager.py", line 922, in install_tool_shed_repository
Feb 21 04:31:56 gcc-6 uwsgi[25895]:     tool_panel_section_mapping=tool_panel_section_mapping)
Feb 21 04:31:56 gcc-6 uwsgi[25895]:   File "/srv/galaxy/server/lib/galaxy/tool_shed/galaxy_install/install_manager.py", line 548, in __handle_repository_contents
Feb 21 04:31:56 gcc-6 uwsgi[25895]:     tool_util.copy_sample_files(self.app, tool_index_sample_files, tool_path=tool_path)
Feb 21 04:31:56 gcc-6 uwsgi[25895]:   File "/srv/galaxy/server/lib/galaxy/tool_shed/util/tool_util.py", line 83, in copy_sample_files
Feb 21 04:31:56 gcc-6 uwsgi[25895]:     copy_sample_file(app, filename, dest_path=dest_path)
Feb 21 04:31:56 gcc-6 uwsgi[25895]:   File "/srv/galaxy/server/lib/galaxy/tool_shed/util/tool_util.py", line 57, in copy_sample_file
Feb 21 04:31:56 gcc-6 uwsgi[25895]:     shutil.copy(full_source_path, full_destination_path)
Feb 21 04:31:56 gcc-6 uwsgi[25895]:   File "/srv/galaxy/venv/lib/python3.6/shutil.py", line 245, in copy
Feb 21 04:31:56 gcc-6 uwsgi[25895]:     copyfile(src, dst, follow_symlinks=follow_symlinks)
Feb 21 04:31:56 gcc-6 uwsgi[25895]:   File "/srv/galaxy/venv/lib/python3.6/shutil.py", line 121, in copyfile
Feb 21 04:31:56 gcc-6 uwsgi[25895]:     with open(dst, 'wb') as fdst:
Feb 21 04:31:56 gcc-6 uwsgi[25895]: PermissionError: [Errno 13] Permission denied: '/srv/galaxy/server/tool-data/bwa_mem_index.loc.sample'
```

This finally worked (except for conda dependencies):

```suggestion
>    6. `tool_data_path` to `{{ galaxy_mutable_data_dir }}/tool-data`, so that when tools are installed, due to privilege separation, this will happen in a directory Galaxy can actually write into.
```

Reply:
We set both `shed_tool_data_path` and `tool_data_path` on EU to the same value but maybe that's a mistake. @natefoo?

Reply:
If not defined `shed_tool_data_path ` falls back to `tool_data_path`, so it isn't necessary but not wrong either.

Reply:
Ok, fixed everywhere to `tool_data_path`